src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -31,6 +31,56 @@
 
 namespace Microsoft.PowerShell.Commands
 {
+    #region InodeTracker
+    /// <summary>
+    /// Tracks visited files/directories by caching their device IDs and inodes.
+    /// </summary>
+    internal class InodeTracker
```

Comment:
This class seems to be a helper that will only be used in FileSystemProvider code, so can we make it a private nested class within FileSystemProvider? (just like `private static class NativeMethods` in FileSystemProvider)